docs/user_guide/assessment.rst

```diff
@@ -815,7 +795,196 @@ For more examples, please
 see the :ref:`sphx_glr_auto_examples_plot_new_metrics.py` notebook in the
 :ref:`examples`.
 
-.. _plot:
+Finally, a :class:`MetricFrame` can use multiple control features, multiple
+sensitive features and multiple metric functions simultaneously.
+
+
+Extra Arguments to Metric functions
+-----------------------------------
+
+The metric functions supplied to :class:`MetricFrame` might require additional
+arguments.
+These fall into two categories: 'scalar' arguments (which affect the operation
+of the metric function), and 'per-sample' arguments (such as sample weights).
+Different approaches are required to use each of these.
+
+Scalar Arguments
+^^^^^^^^^^^^^^^^
+
+We do not directly support scalar arguments for the metric functions.
+If these are required, then use :func:`functools.partial` to prebind the
+required arguments to the metric function:
+
+.. doctest:: assessment_metrics
+    :options:  +NORMALIZE_WHITESPACE
+
+    >>> import functools
+    >>> from sklearn.metrics import fbeta_score
+    >>> fbeta_06 = functools.partial(fbeta_score, beta=0.6)
+    >>> metric_beta = MetricFrame(metrics=fbeta_06,
+    ...                           y_true=y_true,
+    ...                           y_pred=y_pred,
+    ...                           sensitive_features=sf_data)
+    >>> metric_beta.overall
+    0.56983...
+    >>> metric_beta.by_group
+    sensitive_feature_0
+    a    0.365591
+    b    0.850000
+    c    0.468966
+    Name: metric, dtype: float64
+
+
+Per-Sample Arguments
+^^^^^^^^^^^^^^^^^^^^
+
+If there are per-sample arguments (such as sample weights), these can also be 
+provided in a dictionary via the ``sample_params`` argument:
+
+.. doctest:: assessment_metrics
+    :options:  +NORMALIZE_WHITESPACE
+
+    >>> s_w = [1, 2, 1, 3, 2, 3, 1, 2, 1, 2, 3, 1, 2, 3, 2, 3, 1, 1]
+    >>> s_p = { 'sample_weight':s_w }
+    >>> weighted = MetricFrame(metrics=recall_score,
+    ...                        y_true=y_true,
+    ...                        y_pred=y_pred,
+    ...                        sensitive_features=pd.Series(sf_data, name='SF 0'),
+    ...                        sample_params=s_p)
+    >>> weighted.overall
+    0.45...
+    >>> weighted.by_group
+    SF 0
+    a    0.500000
+    b    0.583333
+    c    0.250000
+    Name: recall_score, dtype: float64
+
+If multiple metrics are being evaluated, then ``sample_params`` becomes a 
+dictionary of dictionaries, with the first key corresponding matching that in 
+the dictionary holding the desired underlying metric functions.
```

Comment:
I think this sentence is incomplete?

Reply:
That's a polite way of saying that I wrote gibberish.... ;-)